test/app-tap/CMakeLists.txt

```diff
@@ -6,3 +6,86 @@ target_link_libraries(module_api msgpuck)
 set(CMAKE_C_FLAGS "-Wall -Wextra -std=c99")
 set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Werror")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2")
+
+tarantool_make_lua_path(LUA_PATH
+  PATHS
+    ${CMAKE_CURRENT_SOURCE_DIR}/?.lua
+    ${CMAKE_CURRENT_SOURCE_DIR}/lua/?.lua
+    # Needed for test-run/test_run.lua.
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+tarantool_make_lua_path(LUA_CPATH
+  PATHS
+  ${CMAKE_CURRENT_BINARY_DIR}/?${CMAKE_SHARED_LIBRARY_SUFFIX}
+)
+
+set(TEST_SUITE_NAME "app-tap")
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <app-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS module_api
+)
+
+list(APPEND TEST_ENV
+  "LUA_PATH=${LUA_PATH}"
+  "LUA_CPATH=${LUA_CPATH}"
+)
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -rf ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
Q: Should it be also `${work_dir}` for the consistency?
Here and below.

Reply:
With `WORKING_DIRECTORY` equal to `${work_dir}` test `test/app-tap/gh-2717-no-quit-sigint.test.lua` is failed with `809: LuajitError: ...G/tarantool/test/app-tap/gh-2717-no-quit-sigint.test.lua:265: time_quota is violated`.

Reply:
Nice, thanks!